misc/python/materialize/feature_benchmark/scenario.py

```diff
@@ -46,9 +46,11 @@ def scale(self) -> float:
     def n(self) -> int:
         return self._n
 
+    # TODO cls
+    # TODO num param
     @staticmethod
     def name_with_scale(cls: Type["Scenario"], num: int, params_dict: Dict) -> str:
```

Comment:
@philip-stoev: Do we need the parameter `cls` or can we find an alternative that does not require it? Furthermore, we can remove the param `num`, right?

Reply:
This is used like this

```
@parameterized_class(
    [{"SCALE": i} for i in [5, 6, 7, 8, 9]], class_name_func=Scenario.name_with_scale
)
```

so it is a `@classmethod`. 

The param `num` is passed by the parameterized library, I do not think you can avoid it even if you do not use it.

https://pypi.org/project/parameterized/

> The names of the test cases generated by @parameterized.expand can be customized using the name_func keyword argument. The value should be a function which accepts three arguments: testcase_func, param_num, and params, and it should return the name of the test case. testcase_func will be the function to be tested, param_num will be the index of the test case parameters in the list of parameters, and param (an instance of param) will be the parameters which will be used.